testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveTransactionalTable.java

```diff
@@ -238,6 +238,63 @@ public void testReadInsertOnly(boolean isPartitioned, BucketingType bucketingTyp
         }
     }
 
+    @Test(groups = HIVE_TRANSACTIONAL, dataProvider = "partitioningAndBucketingTypeDataProvider", timeOut = TEST_TIMEOUT)
+    @Flaky(issue = "https://github.com/trinodb/trino/issues/4927", match = "Hive table .* is corrupt. Found sub-directory in bucket directory for partition")
+    public void testReadNonOrcInsertOnly(boolean isPartitioned, BucketingType bucketingType)
+    {
+        if (getHiveVersionMajor() < 3) {
+            throw new SkipException("Hive transactional tables are supported with Hive version 3 or above");
+        }
+
+        try (TemporaryHiveTable table = TemporaryHiveTable.temporaryHiveTable(tableName("insert_only", isPartitioned, bucketingType))) {
+            String tableName = table.getName();
+
+            onHive().executeQuery("CREATE TABLE " + tableName + " (col INT) " +
+                    (isPartitioned ? "PARTITIONED BY (part_col INT) " : "") +
+                    "  ROW FORMAT SERDE\n" +
+                    "    'org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe'\n" +
+                    "  WITH SERDEPROPERTIES (\n" +
+                    "    'field.delim'=',',\n" +
+                    "    'line.delim'='\\n',\n" +
+                    "    'serialization.format'=',')\n" +
+                    "  STORED AS INPUTFORMAT\n" +
+                    "    'org.apache.hadoop.mapred.TextInputFormat'\n" +
```

Comment:
We should cover Parquet as well.

Can there be some reuse between `testReadInsertOnly`? ideally they different in table file format definition?